components/button/test/button.visual-diff.js

```diff
@@ -0,0 +1,82 @@
+const puppeteer = require('puppeteer');
+const VisualDiff = require('@brightspace-ui/visual-diff');
+
+describe('d2l-button', function() {
+
+	const visualDiff = new VisualDiff('button', __dirname);
+
+	let browser, page;
+
+	before(async() => {
+		browser = await puppeteer.launch();
+		page = await browser.newPage();
+		await page.setViewport({width: 800, height: 800, deviceScaleFactor: 2});
+		await page.goto(`${visualDiff.getBaseUrl()}/components/button/test/button.visual-diff.html`, {waitUntil: ['networkidle0', 'load']});
+		await page.bringToFront();
+	});
+
+	after(() => browser.close());
+
```

Comment:
These end up being really repetitive... what do folks thing about using the pattern we did [in icons](https://github.com/BrightspaceUI/core/blob/master/components/icons/test/icon.visual-diff.js#L27)? We'd just need a `focus: true` property.

Reply:
I definitely like that, and was meaning to put together a PR for the others.  Sometimes there ends up being a couple tests that don't "fit" since there's a bit more to the test, we hopefully we can find a nice way to categorize them and reduce the repeated code.

Reply:
@dlockhart @dbatiste I think that'd be something like this? IT's shorty but still feels a bit hacky. what do you think?

```
       [
		{category: 'normal', tests: ['normal', 'hover', 'focus', 'disabled']},
		{category: 'primary', tests: ['normal', 'hover', 'focus', 'primary-disabled']},
	].forEach((entry) => {
		describe(entry.category, () => {
			entry.tests.forEach((name) => {
				it(name, async function() {
					if(name == 'hover') {
						await page.hover(`#${entry.category}`);
					} else if(name == 'focus') {
						await focus(page, `#${entry.category}`);
					}

					let rect = await visualDiff.getRect(page, `#${entry.category}`);;
					if(name.contains('disabled')) {
						rect = await visualDiff.getRect(page, `#${name}`);
					}
					await visualDiff.screenshotAndCompare(page, this.test.fullTitle(), { clip: rect });
				});
			});
		});
	});
```

Reply:
Yeah exactly. I might switch up the one bit to be like this:

```
const rectId = name.contains('disabled') ? name : entry.category;
let rect = await visualDiff.getRect(page, `#${rectId}`);
```
That way it's not getting the rectangle, not finding it and then getting it again.

But yes, like that!